### PR TITLE
7.10.0 blog post

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
https://deploy-preview-2260--babel.netlify.app/blog/2020/05/25/7.10.0

### CODE REVIEW

File: website/blog/2020-05-25-7.10.0.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+This release includes full support for the new `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in), and `@babel/preset-env` now compiles ES2015-style Unicode escapes (`\u{Babe1}`) to the equivalent legacy syntax (`\uDAAA\uDFE1`).
+
+We are also releasing two improvements to the optional chaining operator (`?.`), parser support for the new [Module Attributes proposal](https://github.com/tc39/proposal-module-attributes), and better tree-shaking support for React code.
+
+You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.10.0).
+
+<!-- truncate -->
+
+Alongside with this Babel release, we are releasing the first experimental version of our new polyfills compatibility architecture. We started discussing about this more than one year ago in a [RFC issue](https://github.com/babel/babel/issues/10008) in the Babel repository. As an aside, we now have an official RFC process to discuss about changes that significantly impact the user experience: you can check it out in the [`babel/rfcs`](https://github.com/babel/rfcs) repository!
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:21]:
```suggestion
Alongside this Babel release, we are releasing the first experimental version of our new polyfills compatibility architecture. We began discussions about this over a year ago in a [RFC issue](https://github.com/babel/babel/issues/10008) within the Babel repository. As an aside, we now have an official RFC process for discussing changes that significantly impact our users: you can check it out over in the [`babel/rfcs`](https://github.com/babel/rfcs) repository!
```

File: website/blog/2020-05-25-7.10.0.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  #name;
+  
+  hug(other) {
+    if (#name in other) console.log(`${this.#name} 🤗 {other.#name}`);
+    else console.log("It's not a person!")
+  }
+}
+```
+
+This stage 1 proposal allows to statically check if a given object has a specific private field.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:71]:
```suggestion
This stage 1 proposal allows you to statically check if a given object has a specific private field.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. We can often leverage this characteristic to use a `try`/`catch` statement and detect their absence, but this proposal gives us a more compact and robust syntax to do so.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:73]:
```suggestion
Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. You can detect undefined private fields by leveraging this behavior with a `try`/`catch` statement, but this proposal gives us a more compact and robust syntax to do so.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+You can read more about in the [proposal's description](https://github.com/tc39/proposal-private-fields-in-in/blob/master/README.md).
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:75]:
```suggestion
You can read more about it in the [proposal's description](https://github.com/tc39/proposal-private-fields-in-in/blob/master/README.md).
```

File: website/blog/2020-05-25-7.10.0.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+We also discovered that we still didn't support compiling a 5-years-old ECMAScript feature: `\u{...}`-style Unicode escapes! Thanks to [Justin](https://github.com/jridgewell), `@babel/preset-env` can now compile them by default both in strings and in identifiers.
+
+Lastly, we have added `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-private-methods` to the [`shippedProposals`](https://babeljs.io/docs/en/babel-preset-env#shippedproposals) option of `@babel/preset-env`. These proposals are not Stage 4 (i.e. part of the ECMAScript standard) yet, but they are already enabled by default in [many JavaScript engines](https://github.com/tc39/proposal-class-fields#implementations).
+
+### Improved optional chaining `?.` ergonomics ([#10961](https://github.com/babel/babel/pull/10961), [#11248](https://github.com/babel/babel/pull/11248))
+
+In TypeScript 3.9, the interaction between non-null assertions (postfix `!`) and optional chaining [has been changed](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#parsing-differences-in-optional-chaining-and-non-null-assertions), to make it more useful.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:39]:
```suggestion
In TypeScript 3.9, the interaction between non-null assertions (postfix `!`) and optional chaining [has been changed](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#parsing-differences-in-optional-chaining-and-non-null-assertions) to make it more useful.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+```ts
+foo?.bar!.baz
+```
+
+In TypeScript 3.8 and Babel 7.9, that could would be read as `(foo?.bar)!.baz`: "If `foo` is not nullish, get `.bar` from it. Then trust me that `foo?.bar` is never nullish and always get `.bar` from it". This means that when `foo` is nullish that code would always throw, because we are trying to get `.baz` from `undefined`.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:45]:
```suggestion
In TypeScript 3.8 and Babel 7.9, the above would be read as `(foo?.bar)!.baz`: "If `foo` is not nullish, get `.bar` from it. Then trust me that `foo?.bar` is never nullish and always get `.bar` from it". This means that when `foo` is nullish that code would always throw, because we are trying to get `.baz` from `undefined`.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+In TypeScript 3.9 and Babel 7.10, that code behaves similarly to `foo?.bar.baz`: "If `foo` is not nullish, get `.bar.baz` from it and trust me that `foo?.bar` isn't nullish".
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:47]:
```suggestion
In TypeScript 3.9 and Babel 7.10, the code behaves similarly to `foo?.bar.baz`: "If `foo` is not nullish, get `.bar.baz` from it and trust me that `foo?.bar` isn't nullish".
```

File: website/blog/2020-05-25-7.10.0.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+Additionally, the class fields proposal [recently supports](https://github.com/tc39/proposal-class-fields/pull/301) mixing the optional chaining `?.` with private fields. This means that this code is now valid:
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:49]:
```suggestion
Additionally, the class fields proposal [recently added support](https://github.com/tc39/proposal-class-fields/pull/301) for mixing optional chaining `?.` with private fields. This means that the following code is now valid:
```

File: website/blog/2020-05-25-7.10.0.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+```js
+obj?.property.#priv;
+obj?.#priv;
+```
+
+Note that in the second example, if `obj` is not nullish and it doesn't have the `#priv` field, it would still throw an error (exactly as `obj.#priv` would throw). You can read the next paragraph to see how to avoid it!
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:56]:
```suggestion
Note that in the second example, if `obj` is not nullish and does not have the `#priv` field, it would still throw an error (exactly as `obj.#priv` would throw). You can read the next section to see how to avoid it!
```

File: website/blog/2020-05-25-7.10.0.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+If you or your company want to support Babel and the evolution of JavaScript, but aren't sure how, you can donate to us on our [Open Collective](https://opencollective.com/babel) and, better yet, work with us on the implementation of [new ECMAScript proposals](https://github.com/babel/proposals) directly! As a volunteer-driven project, we rely on the community's support to fund our efforts in supporting the wide range of JavaScript users. Reach out at [team@babeljs.io](mailto:team@babeljs.io) if you'd like to discuss more!
+
+## Highlights
+
+### New features enabled by default
+
+Now that it has reached stage 4, parsing for `import.meta` is enabled by default. Please note that `@babel/preset-env` doesn't support transforming it, because what that object contains is up to the engine and is not defined in the ECMAScript specification.
+
+We also discovered that we still didn't support compiling a 5-years-old ECMAScript feature: `\u{...}`-style Unicode escapes! Thanks to [Justin](https://github.com/jridgewell), `@babel/preset-env` can now compile them by default both in strings and in identifiers.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:33]:
```suggestion
We also discovered that we didn't have support for compiling a 5-year-old ECMAScript feature: `\u{...}`-style Unicode escapes! Thanks to [Justin](https://github.com/jridgewell), `@babel/preset-env` can now compile them in strings and identifiers by default.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. We can often leverage this characteristic to use a `try`/`catch` statement and detect their absence, but this proposal gives us a more compact and robust syntax to do so.
+
+You can read more about in the [proposal's description](https://github.com/tc39/proposal-private-fields-in-in/blob/master/README.md).
+
+You can test this proposal by installing the `@babel/plugin-proposal-private-property-in-object` plugin and adding it to your Babel config.
+
+### Module Attributes parser support ([#10962](https://github.com/babel/babel/pull/10962))
+
+The Modules Attributes proposal allows giving to the engine, to the module loader or to the bundler some additional information about the imported file. For example, you could explicitly specify that it should be parsed as JSON:
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:81]:
```suggestion
The Modules Attributes proposal allows providing the engine, module loader or bundler some additional information about the imported file. For example, you could explicitly specify that it should be parsed as JSON:
```

File: website/blog/2020-05-25-7.10.0.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+```js
+import metadata from "./package.json" with type: "json";
+```
+
+They are also supported also supported in dynamic `import()`, with support for trailing commas to make it easier to add or remove the second parameter:
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:87]:
```suggestion
Additionally, they can also be used with dynamic `import()`. Note the support for trailing commas to make it easier to add or remove the second parameter!
```

File: website/blog/2020-05-25-7.10.0.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+```js
+const metadata = await import(
+  "./package.json",
+  { with: { type: "json" } },
+);
+```
+
+Thanks to [Vivek](https://twitter.com/VivekNayyar09/), Babel now supports parsing these attributes: you can add the `@babel/plugin-syntax-module-attributes` plugin to your Babel config or, if you are using `@babel/parser` directly, you can enable the `moduleAttributes` plugin. Currently, we only accept the `type` attribute but we might relax this restriction in the future depending on how the proposal evolves.
+
+> ℹ️ Babel doesn't _transform_ those attributes, and they should be directly handled by your bundler or by a custom plugin. We are [discussing](https://github.com/airbnb/babel-plugin-dynamic-import-node/issues/92) about whether or not this should be handled by `@babel/plugin-transform-dynamic-import`.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:97]:
```suggestion
> ℹ️ Babel doesn't _transform_ these attributes, and they should be handled directly by your bundler or a custom plugin. We are also [discussing](https://github.com/airbnb/babel-plugin-dynamic-import-node/issues/92) whether or not this should be handled by `@babel/plugin-transform-dynamic-import`.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+### Better tree-shaking for React components ([#11428](https://github.com/babel/babel/pull/11428))
+
+React exposes many pure functions used to annotate or wrap elements, for example `React.forwardRef`, `React.memo` or `React.lazy`. However, minifiers and bundlers aren't aware that these functions are pure and thus they cannot remove them.
+
+Thanks to [Devon](https://twitter.com/devongovett) from the Parcel team, `@babel/preset-react` now injects `/*#__PURE__*/` annotations in those functions calls to mark them as safe to be tree-shaken away.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:103]:
```suggestion
Thanks to [Devon](https://twitter.com/devongovett) from the Parcel team, `@babel/preset-react` now injects `/*#__PURE__*/` annotations in those functions calls to mark them as being safe to be tree-shaken away.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 15]
```diff
@@ -68,23 +68,23 @@ class Person {
 }
 ```
 
-This stage 1 proposal allows to statically check if a given object has a specific private field.
+This stage 1 proposal allows you to statically check if a given object has a specific private field.
 
-Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. We can often leverage this characteristic to use a `try`/`catch` statement and detect their absence, but this proposal gives us a more compact and robust syntax to do so.
+Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. You can detect undefined private fields by leveraging this behavior with a `try`/`catch` statement, but this proposal gives us a more compact and robust syntax to do so.
```
Comment (nicolo-ribaudo) [inline website/blog/2020-05-25-7.10.0.md:120]: I don't like "undefied private fields" here, because given the presence of the `undefined` value in JS it can be easily confused with `foo.#priv === undefined`.

Comment (existentialism) [reply website/blog/2020-05-25-7.10.0.md:121]: Sure, initial intent was to balance "where they aren't defined"...

> You can determine if an object has a particular private field by leveraging...

File: website/blog/2020-05-25-7.10.0.md [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+authorURL: https://twitter.com/NicoloRibaudo
+date:   2020-05-25 0:00:00
+categories: announcements
+share_text: "Babel 7.10.0 Released"
+---
+
+We just released a new minor version of Babel!
+
+This 7.10 release includes:
+- Full support for the new Stage 1 proposal, `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in) (`class A { #a = 1 }`).
```
Comment (JLHwung) [inline website/blog/2020-05-25-7.10.0.md:14]:
```suggestion
- Full support for the new Stage 1 proposal, `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in) (`class A { #prop; static isA(obj) { return #prop in obj } }`).
```

The code example should include brand checks, though it seems more verbose.

File: website/blog/2020-05-25-7.10.0.md [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+```js
+const metadata = await import(
+  "./package.json",
+  { with: { type: "json" } },
+);
+```
+
+Thanks to [Vivek](https://twitter.com/VivekNayyar09/), Babel now supports parsing these attributes: you can add the `@babel/plugin-syntax-module-attributes` plugin to your Babel config or, if you are using `@babel/parser` directly, you can enable the `moduleAttributes` plugin. Currently, we only accept the `type` attribute but we might relax this restriction in the future depending on how the proposal evolves.
+
+> ℹ️ Babel doesn't _transform_ these attributes, and they should be handled directly by your bundler or a custom plugin. We are also [discussing](https://github.com/airbnb/babel-plugin-dynamic-import-node/issues/92) whether or not this should be handled by `@babel/plugin-transform-dynamic-import`.
```
Comment (JLHwung) [inline website/blog/2020-05-25-7.10.0.md:143]:
```suggestion
> ℹ️ Babel doesn't _transform_ these attributes, and they should be handled directly by your bundler or a custom plugin. Currently babel module transformers _ignore_ these attributes. We are [discussing](https://github.com/airbnb/babel-plugin-dynamic-import-node/issues/92) whether we should pass through these attributes in the future.
```

File: website/blog/2020-05-25-7.10.0.md (hunk 16 above)
Comment (nicolo-ribaudo) [reply website/blog/2020-05-25-7.10.0.md:14]: Maybe just `#prop in obj` to keep it short?

File: website/blog/2020-05-25-7.10.0.md (hunk 15 above)
Comment (nicolo-ribaudo) [reply website/blog/2020-05-25-7.10.0.md:121]:
```suggestion
Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. You can determine if an object has a particular private field by leveraging this behavior with a `try`/`catch` statement, but this proposal gives us a more compact and robust syntax to do so.
```

File: website/blog/2020-05-25-7.10.0.md (hunk 16 above)
Comment (nicolo-ribaudo) [reply website/blog/2020-05-25-7.10.0.md:14]:
```suggestion
- Full support for the new Stage 1 proposal, `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in) (`#prop in obj`).
```

Comment (JLHwung) [reply website/blog/2020-05-25-7.10.0.md]: Or we can just delete code example since we have mentioned it before.

File: website/blog/2020-05-25-7.10.0.md [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+authorURL: https://twitter.com/NicoloRibaudo
+date:   2020-05-25 0:00:00
+categories: announcements
+share_text: "Babel 7.10.0 Released"
+---
+
+We just released a new minor version of Babel!
+
+This 7.10 release includes:
+- Full support for the new Stage 1 proposal, `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in) (`#prop in obj`).
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:14]: The second `(`#prop in obj`)` seems redundant?

File: website/blog/2020-05-25-7.10.0.md [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+```
+
+```javascript
+import React from 'react';
+const SomeComponent = /*#__PURE__*/React.lazy(() => import('./SomeComponent'));
+```
+
+### New experimental polyfills architecture ([#10008](https://github.com/babel/babel/issues/10008), [`babel-polyfills`](https://github.com/babel/babel-polyfills))
+
+In the last three years, `@babel/preset-env` has shown its full potential in reducing bundle sizes not only by not transpiling supported syntax features, but also by not including unnecessary `core-js` polyfills.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:163]:
```suggestion
In the last three years, `@babel/preset-env` has helped users reduce bundle sizes by only transpiling the syntax features and including the `core-js` polyfills needed by their target environments.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+- By using `@babel/plugin-transform-runtime`, Babel will inject po<i>n</i>yfills (which are "pure" and don't pollute the global scope) for every used ECMAScript feature supported by `core-js`. This is usually used by library authors.
+
+Our position in the JavaScript ecosystem allows us to push these optimizations even further. `@babel/plugin-transform-runtime` has big advantages for some users over `useBuiltIns`, but it doesn't consider target environments: it's 2020 and probably very few people need to load an `Array.prototype.forEach` polyfill.
+Additionally, why should we limit this ability to automatically inject only the necessary polyfill to `core-js`? There are also DOM polyfills, Intl polyfills, and polyfills for a myriad of other web platform APIs. Additionally, not everyone wants to use `core-js`: there are many other valid ECMAScript polyfills, which have different tradeoffs (e.g. source size vs spec compliancy) and may work better for some of you. For example, we are actively working on an [**`es-shims`**](https://github.com/es-shims/) integration.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:170]:
```suggestion
Additionally, why should we limit this ability to automatically inject only the necessary polyfills to `core-js`? There are also DOM polyfills, Intl polyfills, and polyfills for a myriad of other web platform APIs. Additionally, not everyone wants to use `core-js`: there are many other valid ECMAScript polyfills which have different tradeoffs (e.g. source size vs spec compliancy) and may work better for some of you. For example, we are actively working on an [**`es-shims`**](https://github.com/es-shims/) integration.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+
+What if the logic to inject them was not related to the actual data about the available or required polyfills, so that they can be used and developed independently?
+
+We are now releasing the first experimental version of four new packages: [`babel-plugin-polyfill-regenerator`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-regenerator/README.md), [`babel-plugin-polyfill-corejs2`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs2/README.md), [`babel-plugin-polyfill-corejs3`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs3/README.md) and [`babel-plugin-polyfill-es-shims`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-es-shims/README.md).
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:174]: Might work better with the packages as a bulleted list?

File: website/blog/2020-05-25-7.10.0.md [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+
+These packages all support injecting polyfills both modifying the engine built-ins and by directly importing them without polluting the global scope, as well as also injecting only the polyfills needed only by your browsers.
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:176]: This was hard to parse... maybe something like?

```suggestion
These packages all support a `method` option for adjusting how they're injected (essentially what `@babel/preset-env` and `@babel/plugin-transform-runtime` offer). You can inject a polyfill into an entry point (global scope only) or by usage in your code (both global scope and "pure" options).
```

File: website/blog/2020-05-25-7.10.0.md [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+
+We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors or users to define _their own_ polyfill provider plugins, 
+
+Thanks [Jordan](https://twitter.com/ljharb) for working with [Nicolò](https://twitter.com/NicoloRibaudo) to make it possible to build the [**`es-shims`**](https://github.com/es-shims/) plugin!
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:180]:
```suggestion
Big thanks to [Jordan](https://twitter.com/ljharb) for working with [Nicolò](https://twitter.com/NicoloRibaudo) to make it possible to build the [**`es-shims`**](https://github.com/es-shims/) plugin!
```

File: website/blog/2020-05-25-7.10.0.md [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+
+We are now releasing the first experimental version of four new packages: [`babel-plugin-polyfill-regenerator`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-regenerator/README.md), [`babel-plugin-polyfill-corejs2`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs2/README.md), [`babel-plugin-polyfill-corejs3`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs3/README.md) and [`babel-plugin-polyfill-es-shims`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-es-shims/README.md).
+
+These packages all support injecting polyfills both modifying the engine built-ins and by directly importing them without polluting the global scope, as well as also injecting only the polyfills needed only by your browsers.
+
+We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors or users to define _their own_ polyfill provider plugins,
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:178]:
```suggestion
We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors or users to define _their own_ polyfill provider plugins.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+
+These packages all support injecting polyfills both modifying the engine built-ins and by directly importing them without polluting the global scope, as well as also injecting only the polyfills needed only by your browsers.
+
+We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors or users to define _their own_ polyfill provider plugins, 
+
+Thanks [Jordan](https://twitter.com/ljharb) for working with [Nicolò](https://twitter.com/NicoloRibaudo) to make it possible to build the [**`es-shims`**](https://github.com/es-shims/) plugin!
+
+> ℹ️ If you want to read more about these packages, and learn how to setup them, you can check out the project's [`README`](https://github.com/babel/babel-polyfills/blob/master/README.md).
```
Comment (existentialism) [inline website/blog/2020-05-25-7.10.0.md:182]:
```suggestion
> ℹ️ If you want to read more about these packages, and learn how to set them up, you can check out the project's [`README`](https://github.com/babel/babel-polyfills/blob/master/README.md).
```

File: website/blog/2020-05-25-7.10.0.md (hunk 18 above)
Comment (hzoo) [reply website/blog/2020-05-25-7.10.0.md:14]:
```suggestion
- Full support for the new Stage 1 proposal, `#prop in obj` checks for private fields [proposal](https://github.com/tc39/proposal-private-fields-in-in).
```

File: website/blog/2020-05-25-7.10.0.md [hunk 26]
```diff
@@ -0,0 +1,184 @@
...
+
+If you or your company want to support Babel and the evolution of JavaScript, but aren't sure how, you can donate to us on our [Open Collective](https://opencollective.com/babel) and, better yet, work with us on the implementation of [new ECMAScript proposals](https://github.com/babel/proposals) directly! As a volunteer-driven project, we rely on the community's support to fund our efforts in supporting the wide range of JavaScript users. Reach out at [team@babeljs.io](mailto:team@babeljs.io) if you'd like to discuss more!
+
+## Highlights
+
+### New features enabled by default
+
+Now that it has reached Stage 4, parsing for [`import.meta`](https://github.com/tc39/proposal-import-meta) is enabled by default. Please note that `@babel/preset-env` doesn't support transforming it, because what that object contains is up to the engine and is not defined in the ECMAScript specification.
```
Comment (hzoo) [inline website/blog/2020-05-25-7.10.0.md:35]:
```suggestion
Now that it has reached Stage 4, _parsing_ for [`import.meta`](https://github.com/tc39/proposal-import-meta) is enabled by default. Please note that `@babel/preset-env` doesn't have any default support for _transforming_ it, because what that object contains is up to the engine and is not defined in the ECMAScript specification.
```
aside: if we know the target (node), it should be possible to transform some forms of it to some equivalent in some cases right (`import.meta.url`)?

File: website/blog/2020-05-25-7.10.0.md [hunk 27]
```diff
@@ -0,0 +1,193 @@
...
+- Two improvements to the [Optional Chaining](https://github.com/tc39/proposal-optional-chaining) operator (`?.`)
+- Parser support for the new Stage 1 [Module Attributes proposal](https://github.com/tc39/proposal-module-attributes) (`import a from "./a.json" with type: "json"`).
+- Better tree-shaking support for React code (i.e. `React.memo`)!
+- Setting up [RFCs](https://github.com/babel/rfcs) repo and [GitHub Discussions](https://github.com/babel/babel/discussions) pages!
+
+You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.10.0).
+
+<!-- truncate -->
+
+Alongside this Babel release, we are releasing the first experimental version of our new [polyfills compatibility architecture](https://github.com/babel/babel-polyfills) (see below for more details) thanks to [Nicolò](https://twitter.com/NicoloRibaudo) and some awesome folks in the community! We began discussions about this over a year ago in a [RFC issue](https://github.com/babel/babel/issues/10008) within the Babel repository.
```
Comment (kaicataldo) [inline website/blog/2020-05-25-7.10.0.md:25]:
```suggestion
Alongside this Babel release, we are releasing the first experimental version of our new [polyfills compatibility architecture](https://github.com/babel/babel-polyfills) (see below for more details), thanks to [Nicolò](https://twitter.com/NicoloRibaudo) and some awesome folks in the community! We began discussions about this over a year ago in a [RFC issue](https://github.com/babel/babel/issues/10008) within the Babel repository.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 28]
```diff
@@ -0,0 +1,193 @@
...
+- By using `@babel/plugin-transform-runtime`, Babel will inject po<i>n</i>yfills (which are "pure" and don't pollute the global scope) for every used ECMAScript feature supported by `core-js`. This is usually used by library authors.
+
+Our position in the JavaScript ecosystem allows us to push these optimizations even further. `@babel/plugin-transform-runtime` has big advantages for some users over `useBuiltIns`, but it doesn't consider target environments: it's 2020 and probably very few people need to load an `Array.prototype.forEach` polyfill.
+
+Additionally, why should we limit this ability to automatically inject only the necessary polyfills to `core-js`? There are also DOM polyfills, Intl polyfills, and polyfills for a myriad of other web platform APIs. Additionally, not everyone wants to use `core-js`: there are many other valid ECMAScript polyfills which have different tradeoffs (e.g. source size vs spec compliancy) and may work better for some of you. For example, we are actively working on an [**`es-shims`**](https://github.com/es-shims/) integration.
```
Comment (kaicataldo) [inline website/blog/2020-05-25-7.10.0.md:175]:
```suggestion
Additionally, why should we limit the ability to automatically inject only the necessary polyfills to `core-js`? There are also DOM polyfills, Intl polyfills, and polyfills for a myriad of other web platform APIs. Not everyone wants to use `core-js`; there are many other valid ECMAScript polyfills which have different tradeoffs (e.g. source size versus spec compliancy), and users should have the ability to use the polyfill of their choice. For example, we are actively working on an [**`es-shims`**](https://github.com/es-shims/) integration.
```

File: website/blog/2020-05-25-7.10.0.md [hunk 29]
```diff
@@ -0,0 +1,193 @@
...
+
+What if the logic to inject them was not related to the actual data about the available or required polyfills, so that they can be used and developed independently?
+
+We are now releasing the first experimental version of four new packages:
+- [`babel-plugin-polyfill-regenerator`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-regenerator/README.md)
+- [`babel-plugin-polyfill-corejs2`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs2/README.md)
+- [`babel-plugin-polyfill-corejs3`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-corejs3/README.md)
+- [`babel-plugin-polyfill-es-shims`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-plugin-polyfill-es-shims/README.md).
+
+These packages all support a `method` option for adjusting how they're injected (essentially what `@babel/preset-env` and `@babel/plugin-transform-runtime` offer). You can inject a polyfill into an entry point (global scope only) or by usage in your code (both global scope and "pure" options).
```
Comment (kaicataldo) [inline website/blog/2020-05-25-7.10.0.md:185]:
```suggestion
These packages all support a `method` option for adjusting how they're injected (analogous to what `@babel/preset-env` and `@babel/plugin-transform-runtime` currently offer). You can inject a polyfill into an entry point (global scope only) or by direct usage in your code (both global scope and "pure" options).
```

File: website/blog/2020-05-25-7.10.0.md [hunk 30]
```diff
@@ -0,0 +1,193 @@
...
+
+We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors or users to define _their own_ polyfill provider plugins.
```
Comment (kaicataldo) [inline website/blog/2020-05-25-7.10.0.md:187]:
```suggestion
We are also releasing [`@babel/helper-define-polyfill-provider`](https://github.com/babel/babel-polyfills/blob/master/packages/babel-helper-define-polyfill-provider/README.md): a new helper package which makes it possible for polyfill authors and users to define _their own_ polyfill provider plugins.
```
